the-collab-lab/tcl-79-smart-shopping-list
# Relocate Share List func to modal + navigation to lists from Home + styled Home components

## Description

The share list functionality has been moved to a modal, which opens when the share button next to each list is clicked.
When a list name is clicked, the site navigates to the list view for that specific list.
Components visible from the home page have been styled to match Figma (as much as I could).

## Related Issue

Closes #32 

## Acceptance Criteria

- [x] Each `SingleList` component renders a button that allows the user to open a modal when clicked.
- [x] The modal allos the user to share the list with another user by filling and submitting a form.
- [x] Each `SingleList` component renders a clickable list name, which navigates to that list's view.
- [x] - [x] Each `SingleList` component renders a delete list button that does nothing (for now :))

## Type of Changes

`enhancement`

## Updates

### Before



### After

![image](https://github.com/user-attachments/assets/4dc9cea7-2e86-49a4-a06f-cafaa747c2c3)
![image](https://github.com/user-attachments/assets/effd1be0-d3ca-4064-a982-e30a0955fdbd)


## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout refactor_homepage`
- Open the homepage by running `npm start`.
- The share list functionality should be in a modal, which opens clicking the share icon next to each list.
- When clicking on a list name, the site should navigate to that list's view
